apps/dashboard/src/components/project-navigation/index.js

```diff
@@ -21,24 +21,29 @@ const Tab = {
 };
 
 const ProjectNavigation = ( { activeTab, disableTitleEditor, projectId } ) => {
-	const { saveAndUpdateProject } = useDispatch( STORE_NAME );
+	const { setEditorTitle } = useDispatch( STORE_NAME );
 
-	const project = useSelect(
-		( select ) => select( STORE_NAME ).getProject( projectId ),
+	const projectTitle = useSelect(
+		( select ) => select( STORE_NAME ).getEditorTitle(),
 		[ projectId ]
 	);
 
-	const updateTitle = ( title ) =>
-		saveAndUpdateProject( projectId, {
-			name: title,
-			title,
-		} );
+	const { setEditorContentChanged } = useDispatch( STORE_NAME );
+
+	const updateTitle = ( title ) => {
+		if ( title !== projectTitle ) {
+			setEditorTitle( title );
+			setEditorContentChanged();
```

Comment:
Technically we should be able to omit this and update the reducer to also set the flag on `EDITOR_TITLE_SET`.